File: app/src/main/java/io/github/ovso/blackbox/ui/main/fragment/VideoFragmentPresenter.java
package io.github.ovso.blackbox.ui.main.fragment;

import android.content.DialogInterface;
import android.os.Bundle;
import io.github.ovso.blackbox.data.network.model.SearchItem;

public interface VideoFragmentPresenter {

  void onActivityCreated(Bundle args);

  void onDestroyView();

  void onItemClick(SearchItem data);

  void onLoadMore();

  void onRefresh();

  interface View {

    void setupRecyclerView();

    void refresh();

    void showVideoTypeDialog(DialogInterface.OnClickListener onClickListener);

    void showPortraitVideo(String videoId);

    void showLandscapeVideo(String videoId);

    void showYoutubeUseWarningDialog();

    void setLoaded();

    void setupSwipeRefresh();

    void hideLoading();

    void showLoading();
  }
}

File: app/src/main/java/io/github/ovso/blackbox/ui/main/fragment/VideoFragmentPresenter.kt
package io.github.ovso.blackbox.ui.main.fragment

import androidx.activity.OnBackPressedCallback
import io.github.ovso.blackbox.data.network.model.SearchItem

interface VideoFragmentPresenter {
  fun onDestroyView()
  fun onItemClick(data: SearchItem)
  fun onLoadMore()
  fun onRefresh()
  fun onViewCreated()
  fun onResume()
  fun onPause()

  interface View {
    fun setupRecyclerView()
    fun refresh()
    fun showPortraitVideo(videoId: String)
    fun showLandscapeVideo(videoId: String)
    fun showYoutubeUseWarningDialog()
    fun setLoaded()
    fun changeTitle(title: CharSequence)
    fun showLoading()
    fun hideLoading()
    fun addOnBackPressedDispatcher(onBackPressedCallback: OnBackPressedCallback)
    fun showNativeAdsDialog()
  }
}
